### PR TITLE
Fix sdk error handling

### PR DESCRIPTION
**Pull request by morphis**
## Done

n/a

## QA

- Check out this feature branch
- Run the site using the command `./run serve`
- View the site locally in your web browser at: http://0.0.0.0:8043/
- Run through the following [QA steps](https://canonical-web-and-design.github.io/practices/workflow/qa-steps.html)

## Issue / Card

n/a

## Screenshots

n/a

### CODE REVIEW

File: static/js/anbox-stream-sdk.js [hunk 1]
```diff
@@ -128,25 +128,24 @@ class AnboxStream {
             body: JSON.stringify(details),
         })
         .then(response => {
-            if (response.status !== 200) {
-                this._options.callbacks.error(
-                    new Error("Failed to join instance: " + response.error));
-                return
-            }
+            if (response.status !== 200)
+                throw new Error("Failed to join instance");
```
Comment (aguadoenzo) [inline static/js/anbox-stream-sdk.js]: missing `response.error` message

**Comment by barrymcgee** [comment]
# [Codecov](https://codecov.io/gh/canonical-web-and-design/anbox-cloud.io/pull/67?src=pr&el=h1) Report
> Merging [#67](https://codecov.io/gh/canonical-web-and-design/anbox-cloud.io/pull/67?src=pr&el=desc) into [master](https://codecov.io/gh/canonical-web-and-design/anbox-cloud.io/commit/7c42a5c8bdba0524b97a7016ace1252a67ef20fa&el=desc) will **not change** coverage by `%`.
> The diff coverage is `n/a`.

File: static/js/anbox-stream-sdk.js [hunk 2]
```diff
@@ -165,22 +164,19 @@ class AnboxStream {
             body: JSON.stringify(details),
         })
         .then(response => {
-            if (response.status !== 200) {
-                this._options.callbacks.error(
-                    new Error("Failed to create new instance: " + response.error));
-                return
-            }
+            if (response.status !== 200)
+                throw new Error("Failed to create new instance");
```
Comment (aguadoenzo) [inline static/js/anbox-stream-sdk.js]: missing `response.error` message

File: static/js/anbox-stream-sdk.js (hunk 1 above)
Comment (morphis) [reply static/js/anbox-stream-sdk.js]: Nope, response isn't the json data at this point, it is the raw response object.

File: static/js/anbox-stream-sdk.js (hunk 2 above)
Comment (morphis) [reply static/js/anbox-stream-sdk.js]: See above

File: static/js/anbox-stream-sdk.js [hunk 3]
```diff
@@ -165,22 +164,19 @@ class AnboxStream {
...
+
             return response.json();
         })
         .then(jsonResp => {
-            if (jsonResp.status !== "success") {
-                this._options.callbacks.error(new Error(jsonResp.error));
-                return
-            }
+            if (jsonResp.status !== "success")
+                throw new new Error(jsonResp.error);
```
Comment (carkod) [inline static/js/anbox-stream-sdk.js]: Duplicate `new` keyword

**Pull request merged**